### DISCUSSION TITLE
Any way to synchronously read data?

### DISCUSSION THREAD

**Discussion by azeem-io** [simolus3/drift#3262] [Q&A]
Hi, is there a synchronous version of an operations like this:
```dart
await (select(table)..limit(1)).getSingle()
```

**Comment by simolus3** [comment]
No, drift has no APIs to run queries synchronously. Since some executors are fundamentally asynchronous (like the sqflite based one or the recommended `NativeDatabase.createInBackground`), we can't really offer a synchronous API.
